src/manager.js

```diff
@@ -280,16 +343,34 @@ AdManager.prototype.generateId = function() {
 
 /**
  * Sorts through viewports slot sizes and returns all dimensions as an array.
- * Note, that this function does NOT filter out sizes that aren't able to display based on the viewport dimensions in the ad unit config.
+ * Note, that this function filters out sizes that aren't able to display based on the viewport dimensions in the ad unit config.
 
  * @param {Array} Viewport size specifications for the ad slot, and list of eligbile sizes for each.
  * @returns {Array} An array of ad sizes belonging to the slot
 */
 
-AdManager.prototype.adUnitSizes = function(adUnitSizes) {
-  return adUnitSizes.filter(function (sizes) {
-    return sizes[1];
-  })[0];
+AdManager.prototype.adUnitSizes = function(sizeMap) {
+  var validSizesIndex = 0;
+  var sizeMapWidths = sizeMap.map(function(sizing, mapIndex) {
+    return [sizing[0][0], mapIndex];
+  }).sort(function(a, b) {
+    return a[0] - b[0];
+  }).forEach(function(element) {
+    if (element[0] <= window.innerWidth) {
+      validSizesIndex = element[1];
+    }
+  });
+  return sizeMap[validSizesIndex][1];
+};
+
+AdManager.prototype.buildSizeMap = function(sizes) {
```

Comment:
This was added because of a quirk in gpt sizemaps.  The order actually matters for the 'media queries' here, so we can rely on the gpt sizemap builder to order them for us.

Noticed the issue on reductress mobile-sticky unit where 320x50 would get sent to DFP despite the 728x90 being the valid size.

Media query order should be  descending widths